File: seed/php-sdk/validation/src/Types/Shape.php
<?php

namespace Seed\Types;

use JsonSerializable;

enum Shape: string implements JsonSerializable
{
    case Square = "SQUARE";
    case Circle = "CIRCLE";
    case Triangle = "TRIANGLE";

    /**
     * @return string
     */
    public function jsonSerialize(): string
    {
        return $this->value;
    }
}

File: seed/ruby-sdk-v2/unions-with-local-date/lib/seed/union/types/shape.rb
# frozen_string_literal: true

module Seed
  module Union
    module Types
      class Shape < Internal::Types::Model
        extend Seed::Internal::Types::Union

        discriminant :type

        member -> { Seed::Union::Types::Circle }, key: "CIRCLE"
        member -> { Seed::Union::Types::Square }, key: "SQUARE"
      end
    end
  end
end
